### PR TITLE
Introduce a max limit on writes to the ProtobufDescriptor table in 1 TX during Snapshot Sync

### PR DESCRIPTION
**Pull request by pankti-m**
Snapshot Writer coalesces all updates to a table into a single transaction with the max size of 25MB(max supported write size).  Applications running with a small memory footprint cannot deserialize a transaction of this size.  So introduce another limit of max number of entries applied in a transaction.

## Overview

Description:

Why should this be merged: This was hit during Upgrade testing.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ public class LogReplicationFSMTest extends AbstractViewTest implements Observer
     private static final int CORFU_PORT = 9000;
     private static final int TEST_TOPOLOGY_CONFIG_ID = 1;
     private static final String TEST_LOCAL_CLUSTER_ID = "local_cluster";
+    private static final int MAX_SNAPSHOT_ENTRIES_APPLIED = 50;
```
Comment (hisundar) [inline test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java:75]: can we have a public static constant defined to be used here as well as the ServerContext area?

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java [hunk 2]
```diff
@@ -89,16 +89,22 @@ public class LogReplicationConfig {
      */
     private int maxDataSizePerMsg;
 
+    /**
+     * Max number of entries to be applied during a snapshot sync.  For special tables only.
+     */
+    private int maxSnapshotEntriesApplied = 50;
```
Comment (hisundar) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java:95]: can we have a public static constant defined to be used here?

File: test/src/test/java/org/corfudb/integration/LogReplicationReaderWriterIT.java [hunk 3]
```diff
@@ -69,6 +69,7 @@ public class LogReplicationReaderWriterIT extends AbstractIT {
     private static final String PRIMARY_SITE_ID = "Cluster-Paris";
     private static final String SHADOW_SUFFIX = "_SHADOW";
     private static final String TEST_NAMESPACE = "LR-Test";
+    private static final int MAX_SNAPSHOT_ENTRIES_APPLIED = 50;
```
Comment (hisundar) [inline test/src/test/java/org/corfudb/integration/LogReplicationReaderWriterIT.java:72]: can we have a public static constant defined to be used here?

**Review by hisundar — COMMENTED** [review]
Looks good overall. thanks

**Review by Lujie1996 — APPROVED** [review]

**Review by hisundar — APPROVED** [review]

**Pull request merged**